drv/fpga-app-api/src/power_rail.rs

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use drv_fpga_api::FpgaError;
+use userlib::FromPrimitive;
+use zerocopy::{AsBytes, FromBytes};
+
+/// This module implements state primitives matching the generic power
+/// rail/voltage regulator as implemented in
+/// https://github.com/oxidecomputer/quartz/blob/main/hdl/power_rail.rdl,
+/// https://github.com/oxidecomputer/quartz/blob/main/hdl/PowerRail.bsv.
```

Comment:
We have a pattern to tightly integrate types defined via RDL into our BSV/Rust code. At first blush, leaning on that could easily eliminate the duplicate hand coding of the `PowerRailStatus` enum. I don't recall if we support generating structs off hand, so `PowerFailPinState` may not be covered but that seems like a natural place to continue to expand our integration here lest we need to change RDL, BSV, and Rust every time a register changes.

Reply:
Yes, the reason this looks the way it does is because while RDL can define types, we currently can't export those types through the JSON representation into Hubris. We can only generate specific instantiations of that type within a register map. We should figure out how to extend our capabilities here, but I don't want to hold up this diff on something open-ended.